checker/src/main/java/org/checkerframework/checker/resourceleak/MustCallConsistencyAnalyzer.java

```diff
@@ -326,11 +495,15 @@ private boolean isValidInvocation(
         Obligation toRemove = null;
         Obligation toAdd = null;
         for (Obligation obligation : obligations) {
+          // It seems that this predicate could be made into a method in the `Obligation` class,
+          // which would make the code slightly shorter and -- more importantly -- clearer.
           for (ResourceAlias alias : obligation.resourceAliases) {
             if (expression.equals(alias.reference)) {
               // This satisfies case 2 above. Remove all its aliases, then return below.
               if (toRemove != null) {
-                throw new BugInCF(
+                // TODO: I changed this from BugInCF to TypeSystemError.  Please look over the other
+                // uses of BugInCF and use TypeSystemError instead where appropriate.
+                throw new TypeSystemError(
```

Comment:
@mernst I don't think the use of `TypeSystemError` is appropriate. It's documentation says that it should only be thrown by the `framework` subproject, and that it indicates that a type system mis-used the framework.

That said, `BugInCF`'s documentation directly contradicts that. In my experience, we've always used `BugInCF` to indicate an error in a checker that's part of the framework. Maybe you're trying to change that, but if so can you please clarify the documentation on the two errors so that it is clear to type system implementers which to use? Until then, I will continue to use `BugInCF`.

Reply:
`TypeSystemError` is relatively new (just over a year old).  We are trying to distinguish between bugs in the framework and bugs in checkers built on the framework, and it is a way to do that.  There may be uses of `BugInCF` that still need to be changed to `TypeSystemError`.  I removed the remark about "only thrown by the `framework` subproject in #4810.  Let me know if it is clearer, or it should be enhanced.